config/config.go

```diff
@@ -1,20 +1,65 @@
 package config
 
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrMultipleAuth   = errors.New("only one auth. option can be used (API key or WCS)")
+	ErrIncompleteAuth = errors.New("authentication info incomplete")
+)
+
 type Config struct {
 	// Host of the Weaviate instance.
 	Endpoint string `json:"endpoint" validate:"required"`
 
 	// Scheme of the Weaviate instance.
 	Scheme string `json:"scheme" default:"https" validate:"inclusion=http|https"`
 
-	//TODO: support additional auth schemes __sL__
-	// https://github.com/conduitio-labs/conduit-connector-weaviate/issues/4
-
 	// A Weaviate API key
 	APIKey string `json:"apiKey"`
 
+	// Weaviate Cloud Services (WCS) credentials.
+	WCS WCSAuth `json:"wcs"`
+
 	// The class name as defined in the schema.
 	// A record will be saved under this class unless
 	// it has the `weaviate.class` metadata field.
 	Class string `json:"class" validate:"required"`
 }
+
+func (c *Config) Validate() error {
```

Comment:
what do you think about having a config value to choose `auth.method` and according to the method, we either use the `auth.username` and `auth.password` values, or the `auth.apiKey`, and we don't care if both of them is set or not, we just ignore what's not specified as `auth.method`, I think we do this approach in many other connector

Reply:
Sure, I think it makes sense to have a common approach. Just so that different auth. options stand out, wdyt about following:
* `auth.method` can be `none`, `apiKey` or `wcsCredentials`
* If `auth.method = apiKey` then you need to provide `auth.apiKey`
* If `auth.method = wcsCredentials`, then you need to provide `auth.wcsCredentials.username` and `auth.wcsCredentials.password`.

Reply:
hmm what about `auth.WCS` or `auth.wcs` instead of `auth.wcsCredentials`?
also, my bad but I think we use `auth.mechanism` not `auth.method`, even though they both sound ok

Reply:
`auth.wcs` is a good option as well and also shorter then `auth.wcsCredentials`. The only reason I put `wcsCredentials` is because WCS itself means Weaviate Cloud Services, which isn't an authentication method itself.

Reply:
then maybe `auth.wcsCreds`?

Reply:
Sounds good... But it's still a problem because of something else. Namely, using a dot in a json tag doesn't work that well. E.g. this fails:
```go

type Config struct {
	AConfig AnotherConfig `json:"another.config"`

	Param1 string
}

type AnotherConfig struct {
	AnotherParam string `json:"anotherParam"`
}

func TestConfig_Parse(t *testing.T) {
	cfg := Config{}
	err := sdk.Util.ParseConfig(
		map[string]string{
			"another.config.anotherParam": "foo",
		},
		&cfg,
	)
	is := is.New(t)
	is.NoErr(err)
	is.Equal("foo", cfg.AConfig.AnotherParam)
}
```

Reply:
Here's what I did in the end: https://github.com/conduitio-labs/conduit-connector-weaviate/pull/6/commits/14677d03c9a4e65b1b52d3e43ed8d4a52ee3280d

Reply:
oh I see, to solve this, I think it would be nicer to have a struct called `Authentication` or `Auth` which will contain all the auth related values and add that to the config struct, so `paramgen` will take care of adding the `dot` between them.
```
type Config struct {
        OtherConfigs vals
	Authentication  auth `json:"auth"`
}
```

Reply:
Ah, such a simple and elegant solution, don't know why I didn't consider it before. Makes a lot of sense to me.